optuna/_convert_positional_args.py

```diff
@@ -37,22 +61,66 @@ def convert_positional_args(
     *,
     previous_positional_arg_names: Sequence[str],
     warning_stacklevel: int = 2,
+    deprecated_version: str | None = None,
+    removed_version: str | None = None,
 ) -> "Callable[[Callable[_P, _T]], Callable[_P, _T]]":
     """Convert positional arguments to keyword arguments.
 
     Args:
-        previous_positional_arg_names: List of names previously given as positional arguments.
-        warning_stacklevel: Level of the stack trace where decorated function locates.
+        previous_positional_arg_names:
+            List of names previously given as positional arguments.
+        warning_stacklevel:
+            Level of the stack trace where decorated function locates.
+        deprecated_version:
+            Version number in which the use of positional arguments is deprecated.
+        removed_version:
+            Version number in which the use of positional arguments was completely removed.
```

Comment:
It would be better to align these explanations with `deprecated_func`:
```suggestion
        deprecated_version:
            The version in which the use of positional arguments is deprecated.
        removed_version:
            The version in which the use of positional arguments will be removed.
```